plugin/src/main/java/org/opensearch/ml/task/MLTrainAndPredictTaskRunner.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ *
+ */
+
+package org.opensearch.ml.task;
+
+import static org.opensearch.ml.plugin.MachineLearningPlugin.TASK_THREAD_POOL;
+import static org.opensearch.ml.stats.StatNames.ML_EXECUTING_TASK_COUNT;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionListenerResponseHandler;
+import org.opensearch.action.support.ThreadedActionListener;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.ml.common.dataframe.DataFrame;
+import org.opensearch.ml.common.dataset.DataFrameInputDataset;
+import org.opensearch.ml.common.dataset.MLInputDataType;
+import org.opensearch.ml.common.parameter.MLInput;
+import org.opensearch.ml.common.parameter.MLOutput;
+import org.opensearch.ml.common.parameter.MLPredictionOutput;
+import org.opensearch.ml.common.transport.prediction.MLPredictionTaskResponse;
+import org.opensearch.ml.common.transport.training.MLTrainingTaskRequest;
+import org.opensearch.ml.common.transport.trainpredict.MLTrainAndPredictionTaskAction;
+import org.opensearch.ml.engine.MLEngine;
+import org.opensearch.ml.indices.MLInputDatasetHandler;
+import org.opensearch.ml.model.MLTask;
+import org.opensearch.ml.model.MLTaskState;
+import org.opensearch.ml.model.MLTaskType;
+import org.opensearch.ml.stats.MLStats;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+/**
+ * MLPredictTaskRunner is responsible for running predict tasks.
+ */
+@Log4j2
+public class MLTrainAndPredictTaskRunner extends MLTaskRunner<MLTrainingTaskRequest, MLPredictionTaskResponse> {
+    private final ThreadPool threadPool;
+    private final ClusterService clusterService;
+    private final Client client;
+    private final MLInputDatasetHandler mlInputDatasetHandler;
+
+    public MLTrainAndPredictTaskRunner(
```

Comment:
I think we will need add unit tests for these new classes.

Reply:
yes, will add more test once refactor done.